### PR TITLE
Off hand item drop fix

### LINKED ISSUE #48966: Long range TK with binoculars trick
## Round ID:
Not a /tg/ round (I found this trick on fulpstation, but it should still work over on /tg/).

## Testmerges:
Unknown; probably not relevant.

## Reproduction:

Use telekinesis to pick up binoculars from a distance, then press "z" with them in (telekinetic) hand to use the binoculars to expand your vision range (until you move)... even though that wouldn't logically work.

But wait, there's more! If you, without leaving your tile, then stop telekinetically holding the binoculars (or alternatively switch to your (occupied) off hand and press q), your other hand will still be occupied, but you'll have a free hand and keep the expanded vision range until you leave your tile. You can then use your free hand to use telekinesis on things that you can only see with this expanded vision range, which can be quite useful for telegriefing people from maintenance over a screen away. Note that there _is_ a maximum range for telekinetically gripping an object, it just happens to be longer than your normal viewing distance. Note that you _might_ have to stand on the same tile as your telekinetically held binoculars for this trick to work; I forgot the specifics.

tldr;
1. use TK on binoculars
2. use them in hand
3. drop them
4. keep your expanded vision range (until you move) and use it to perform long-range TK(+x-ray+thermal vision) shenanigans
5. also use it to deal some damage to a space dragon while remaining safe and completely hidden from it in maint

### LINKED ISSUE #49722: Synth banana and broom can drop "off hand" symbol
Issue reported from  Round ID: 131518 (/tg/Station Manuel [ENGLISH] [US-WEST] [ROLEPLAY] [99% LAG FREE])

Reporting client version: 513.1507

"OFF-HAND" symbol appears to be "drop-able" during certain conditions.

##Reproduction
Here is the method as given by the user it occurred too:
-Create synthetic banana skin from honkmother staff
-Dual wield the broom, so that it is in both hands
-Telepathically slip person who is dual wielding the broom
-When they slip they drop the broom AND the "OFF-HAND" symbol as well.

Pictures:
Image of offhand

Image of right click

### PR DESCRIPTION
**Pull request by nightred**
## About The Pull Request

Made the off hand item have the no drop trait and fixed up off hand item deleting the wrong object.

Fixes #49722
Fixes #48966

## Why It's Good For The Game

The offhand item should not be droppable

## Changelog
:cl:
fix: offhand placeholder should not be droppable
fix: no more holding objects in two hands with your mind
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
Aw, does this fix https://github.com/tgstation/tgstation/issues/48966 ?

File: code/datums/components/twohanded.dm [hunk 1]
```diff
@@ -117,7 +117,7 @@
  * * user The mob/living/carbon that is wielding the item
  */
 /datum/component/two_handed/proc/wield(mob/living/carbon/user)
-	if(wielded)
+	if(wielded || !istype(user))
```
Comment (ninjanomnom) [inline code/datums/components/twohanded.dm]: Why are non carbons being passed in?

File: code/datums/components/twohanded.dm [hunk 2]
```diff
@@ -266,6 +266,11 @@
 		return
 	if(held_item == parent)
 		return COMPONENT_BLOCK_SWAP
+	// check if we are holding an item with telekinesis, is that item our parent?
+	if(istype(held_item, /obj/item/tk_grab))
+		var/obj/item/tk_grab/tk_item = held_item
+		if (tk_item?.focus == parent)
```
Comment (ninjanomnom) [inline code/datums/components/twohanded.dm]: tk_item definitely exists, you istyped it on line 270
```suggestion
		if (tk_item.focus == parent)
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/components/twohanded.dm (hunk 2 above)
Comment (nightred) [reply code/datums/components/twohanded.dm]: This was actually the first iteration of blocking TK from using items two handed.
It was replaced and i never removed this, this block is now gone.

File: code/datums/components/twohanded.dm (hunk 1 above)
Comment (nightred) [reply code/datums/components/twohanded.dm]: this run timed on the first iteration of the TK block, it passed in the turf.
This issue is now resolved by prevent TK from wielding two handed items.

**Pull request merged**